ui/plugins/eu.esdihumboldt.hale.ui.filter/src/eu/esdihumboldt/hale/ui/filter/cql/CQLFilterField.java

```diff
@@ -95,11 +96,18 @@ protected String selectVariable() {
 				&& dialog.getObject().getType().getName().toString().length() >= 1) {
 			StringBuilder var = new StringBuilder();
 			for (int i = 0; i < dialog.getObject().getPropertyPath().size(); i++) {
-				if (i != 0)
-					var.append(".");
-				var.append(dialog.getObject().getPropertyPath().get(i).getChild().getName()
-						.getLocalPart().toString());
+				ChildContext child = dialog.getObject().getPropertyPath().get(i);
+
+				if (child.getChild().asGroup() == null) {
+					if (i != 0)
+						var.append(".");
```

Comment:
> The case that you don't handle here however is that the first element (index 0) could have been a group/choice. In that case you would also not want to add a dot.

This is not solved yet.
If the  first element with index 0 was a group/choice then you would start the string with a dot, which is not correct.